core/src/test/scala/unit/kafka/log/LogValidatorTest.scala

```diff
@@ -38,32 +38,46 @@ class LogValidatorTest {
 
   @Test
   def testOnlyOneBatchCompressedV0(): Unit = {
-    checkOnlyOneBatchCompressed(RecordBatch.MAGIC_VALUE_V0, CompressionType.GZIP)
+    checkOnlyOneBatch(RecordBatch.MAGIC_VALUE_V0, CompressionType.GZIP)
   }
 
   @Test
   def testOnlyOneBatchCompressedV1(): Unit = {
-    checkOnlyOneBatchCompressed(RecordBatch.MAGIC_VALUE_V1, CompressionType.GZIP)
+    checkOnlyOneBatch(RecordBatch.MAGIC_VALUE_V1, CompressionType.GZIP)
   }
 
   @Test
   def testOnlyOneBatchCompressedV2(): Unit = {
-    checkOnlyOneBatchCompressed(RecordBatch.MAGIC_VALUE_V2, CompressionType.GZIP)
+    checkOnlyOneBatch(RecordBatch.MAGIC_VALUE_V2, CompressionType.GZIP)
+  }
+
+  @Test
+  def testAllowMultiBatchUncompressedV0(): Unit = {
+    checkAllowMultiBatch(RecordBatch.MAGIC_VALUE_V0, CompressionType.NONE)
```

Comment:
Hmm... I'm not sure this is hitting the case we want. We also want to verify the case when the messages go from uncompressed to compressed. Maybe `checkAllowMultiBatch` can accept both source and target codec?